libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -543,8 +623,31 @@ void AP_Baro::update(void)
             }
             // sanity check altitude
             sensors[i].alt_ok = !(isnan(altitude) || isinf(altitude));
+
             if (sensors[i].alt_ok) {
-                sensors[i].altitude = altitude + _alt_offset_active;
+                altitude += _alt_offset_active;
+
+                if (process_gps_alt) {
+                    // calculation of actual error based on current and calibrated GPS altitude
+                    const float gps_err = gps_alt - _gps_calibration_altitude - altitude;
+
+                    // limitation of error magnitude
+                    float prev_err = _gps_alt_error[i].get();
+                    const float constr_gps_err_diff = constrain_value(gps_err - prev_err, -BARO_GPS_MAX_ERR, BARO_GPS_MAX_ERR);
```

Comment:
Can you explain why we need this if the rate of change of error is already constrained by line 640? This appears to be constraining the change in error, not the absolute value as suggested by the comment.